src/Compilers/Core/Portable/SourceGeneration/GeneratorDriver.cs

```diff
@@ -183,6 +122,153 @@ public GeneratorDriver AddAdditionalTexts(ImmutableArray<AdditionalText> additio
+            var stateBuilder = ArrayBuilder<GeneratorState>.GetInstance();
+            int receiverCount = 0;
+
+            for (int i = 0; i < state.Generators.Length; i++)
+            {
+                var generator = state.Generators[i];
+                var generatorState = state.GeneratorStates[i];
+
+                // initialize the generator if needed
+                if (!generatorState.Info.Initialized)
+                {
+                    InitializationContext context = new InitializationContext(cancellationToken);
+                    Exception? ex = null;
+                    try
+                    {
+                        generator.Initialize(context);
+                    }
+                    catch (Exception e)
+                    {
+                        ex = e;
+                    }
+                    generatorState = ex is null
+                                     ? new GeneratorState(context.InfoBuilder.ToImmutable())
+                                     : SetGeneratorException(new GeneratorState(), generator, ex, diagnosticsBag, isInit: true);
+                }
+
+                // create the syntax receiver if requested
+                if (generatorState.Info.SyntaxReceiverCreator is object)
+                {
+                    ISyntaxReceiver? rx = null;
+                    try
+                    {
+                        rx = generatorState.Info.SyntaxReceiverCreator();
+                    }
+                    catch (Exception e)
+                    {
+                        generatorState = SetGeneratorException(generatorState, generator, e, diagnosticsBag);
+                    }
+
+                    if (rx is object)
+                    {
+                        generatorState = generatorState.WithReceiver(rx);
+                        receiverCount++;
+                    }
+                }
+
+                stateBuilder.Add(generatorState);
+            }
+
+            // Run a syntax walk if any of the generators requested it
+            if (receiverCount > 0)
+            {
+                for (int i = 0; i < stateBuilder.Count; i++)
+                {
+                    var generatorState = stateBuilder[i];
+                    if (generatorState.SyntaxReceiver is object)
+                    {
+                        GeneratorSyntaxWalker walker = new GeneratorSyntaxWalker(generatorState.SyntaxReceiver);
+                        foreach (var tree in compilation.SyntaxTrees)
+                        {
+                            var root = tree.GetRoot(cancellationToken);
+                            try
+                            {
+                                walker.Visit(root);
+                            }
+                            catch (Exception e)
+                            {
+                                stateBuilder[i] = SetGeneratorException(generatorState, state.Generators[i], e, diagnosticsBag);
```

Comment:
Should we continue attempting to use the generator if it faulted? It seems like this would be undefined state for the receiver?

Reply:
We don't. See line 247

---
In reply to: [466074547](https://github.com/dotnet/roslyn/pull/46296#discussion_r466074547) [](ancestors = 466074547)

Reply:
But we'll still have their walkers in this loop, right? And still call them back.

---
In reply to: [466081752](https://github.com/dotnet/roslyn/pull/46296#discussion_r466081752) [](ancestors = 466081752,466074547)

Reply:
Not addressed?

---
In reply to: [466687438](https://github.com/dotnet/roslyn/pull/46296#discussion_r466687438) [](ancestors = 466687438,466081752,466074547)

Reply:
Oh, yea because we've flipped the order that's now true. 

---
In reply to: [467212217](https://github.com/dotnet/roslyn/pull/46296#discussion_r467212217) [](ancestors = 467212217,466687438,466081752,466074547)